remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/matchers/Result.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.remote.messaging.impl.matchers
+
+sealed class Result {
+    object Match : Result()
+    object Fail : Result()
+    object NextMessage : Result()
+}
+
+fun String.defaultValue() = this.isEmpty()
+fun Int.defaultValue() = this == -1
+
+fun Boolean?.toResult(): Result {
+    return when (this) {
+        true -> Result.Match
+        false -> Result.Fail
+        null -> Result.NextMessage
+    }
```

Comment:
I think we should mark all these as `internal`. We don't want extension funs on primitive types that might be made visible across the board

Even better, I would move this extension funs as private inside the `MatchingAttribute` class